base/reduce.jl

```diff
@@ -96,16 +96,16 @@ foldl(op, itr) = mapfoldl(identity, op, itr)
 
 ## foldr & mapfoldr
 
-function mapfoldr_impl(f, op, v0, itr, i::Integer)
+function mapfoldr_impl(f, op, v0, itr_r, i)
```

Comment:
Thanks for tackling this! I was wondering if there is any difference between `mapreduce`  and `mapfoldr_impl`?

```
mapfoldr(f, op, v0, itr) = mapreduce(f, op, v0 , Iterators.reverse(itr))
```

should be equivalent.

Reply:
Two problems with this.

1. `mapreduce` doesn't define its associativity — it is allowed to associate however it wants.
2. even if you use `mapfoldl`, you are assuming that `op` is commutative as well as associative.  `foldr` changes associativity, but does not commute operands.

Reply:
There is a difference, if I am not mistaken.

```
julia> import Base.mapfoldr; mapfoldr(f, op, v0, itr) = mapfoldl(f, op, v0 , Iterators.reverse(itr))
mapfoldr (generic function with 2 methods)

julia> mapfoldr(identity, /, 1, 1:3)
0.16666666666666666
```
But the correct value should be 1.5.